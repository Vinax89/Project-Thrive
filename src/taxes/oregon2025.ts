export function estimateOregonTax(income: number) {
  const brackets = [
    { upTo: 4400, rate: 0.0475 },
    { upTo: 11000, rate: 0.0675 },
    { upTo: 125000, rate: 0.0875 },
    { upTo: Infinity, rate: 0.099 }
  ];
  let tax = 0, prev = 0;
  for (const b of brackets) {
    const amt = Math.max(0, Math.min(income, b.upTo) - prev);
    tax += amt * b.rate;
    prev = b.upTo;
    if (income <= b.upTo) break;
  }
  return Math.round(tax * 100) / 100;
}
