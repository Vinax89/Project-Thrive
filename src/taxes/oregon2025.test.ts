import { describe, it, expect } from 'vitest';
import { estimateOregonTax } from './oregon2025';

describe('estimateOregonTax', () => {
  const boundaries = [4400, 11000, 125000];

  it('calculates tax at bracket boundaries', () => {
    const taxes = boundaries.map((income) => estimateOregonTax(income));
    expect(taxes).toMatchSnapshot();
  });

  it('returns 0 for no income', () => {
    expect(estimateOregonTax(0)).toBe(0);
  });

  it('calculates tax in the top bracket', () => {
    expect(estimateOregonTax(1_000_000)).toBe(97254.5);
  });
});
