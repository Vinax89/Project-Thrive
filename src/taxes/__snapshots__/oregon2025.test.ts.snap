// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`estimateOregonTax > calculates tax at bracket boundaries 1`] = `
[
  209,
  654.5,
  10629.5,
]
`;
